addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -7468,6 +7505,41 @@ registry.ImageTools = ImageHandlerOption.extend({
             this._requestUserValueWidgets("remove_img_shape_opt")[0].enable();
         }
     },
+    /**
+     * @override
+     */
+    async _select(previewMode, widget) {
+        await this._super(...arguments);
+        // This is a special case where we need to override the "_select"
+        // function in order to trigger mouse events for hover effects on the
+        // images when previewing the options. This is done here because if it
+        // was done in one of the widget methods, the animation would be
+        // canceled when "_refreshPublicWidgets" is executed in the "_super"
+        if (widget.$el[0].closest("#o_hover_effects_options")) {
+            const hasSetImgShapeHoverEffectMethod = widget.getMethodsNames().includes("setImgShapeHoverEffect");
+            // We trigger the animation when preview mode is "false", except for
+            // the "setImgShapeHoverEffect" option, where we trigger it when
+            // preview mode is "true".
+            if (previewMode === hasSetImgShapeHoverEffectMethod) {
+                // The first 'setTimeout' prevents the triggering of animations
+                // when quickly hovering over multiple options.
+                const hoverEffectName = this.$target[0].dataset.hoverEffect;
+                setTimeout(() => {
+                    if (hoverEffectName === this.$target[0].dataset.hoverEffect) {
+                        this.$target[0].dispatchEvent(new Event("mouseover"));
+                        this.mouseoutDispatched = false;
+                        setTimeout(() => {
+                            if (!this.mouseoutDispatched && (hoverEffectName ===
+                                    this.$target[0].dataset.hoverEffect)) {
+                                this.mouseoutDispatched = true;
+                                this.$target[0].dispatchEvent(new Event("mouseout"));
+                            }
+                        }, 700);
+                    }
+                }, 300);
```

Comment:
> If I understand correctly, you use the first setTimeout to wait for a bit before triggering the animation, right ?

Yes, that's correct. Without it, some previews are launched unnecessarily when passing by to reach an option that is not adjacent to the previous one.

> But I have a question: why is the first `setTimeout` necessary ? When I test without it, I don't have the issue

I tested it and encountered the issue when quickly hovering over all the options in the select. It's not consistent, but it happens occasionally. Thank you for your suggestion. I will test that :+1:

Reply:
In fact, it seems like this issue mainly occurs when rapidly switching between options multiple times, which is actually an uncommon use case and shouldn't happen. In this case, I propose removing the first `setTimeout` if you're okay with it. This will help keep the code simple without adding unnecessary complexity.

Reply:
Hmm I would say that if the issue doesn't happen when going from "Overlay" to "Mirror Blur" and the other way around (by hovering the ones between them) and if it only happens if we do that really quickly and multiple times (like when testing), then in that case we could ignore the issue and therefore remove the first `setTimeout` indeed (it would be a nitpick in that case) :thinking: 

So yes, since it didn't happen anymore (as I said) or only occasionally (like you said), we can ignore it :slightly_smiling_face: After all, it is not breaking anything so at worst we could just fix that later if someone complains :wink: 

(just in case, what I suggested was just something like this :arrow_down:  (still not tested) 
(-> not too much complexity in my opinion :wink: )):
```js
/**
 * @override
 */
async _select(previewMode, widget) {
    await this._super(...arguments);
    // This is a special case where we need to override the "_select"
    // function in order to trigger mouse events for hover effects on the
    // images when previewing the options. This is done here because if it
    // was done in one of the widget methods, the animation would be
    // canceled when "_refreshPublicWidgets" is executed in the "_super"
    if (widget.$el[0].closest("#o_hover_effects_options")) {
        const hasSetImgShapeHoverEffectMethod = widget.getMethodsNames().includes("setImgShapeHoverEffect");
        // We trigger the animation when preview mode is "false", except for
        // the "setImgShapeHoverEffect" option, where we trigger it when
        // preview mode is "true".
        if (previewMode === hasSetImgShapeHoverEffectMethod) {
            if (hoverEffectName === this.$target[0].dataset.hoverEffect) {
                this.$target[0].dispatchEvent(new Event("mouseover"));
                this.hoverTimeoutId = setTimeout(() => {
                    if (hoverEffectName === this.$target[0].dataset.hoverEffect) {
                        this.$target[0].dispatchEvent(new Event("mouseout"));
                    }
                }, 700);
            }
        } else if (previewMode === "reset") {
            clearTimeout(this.hoverTimeoutId);
            this.$target[0].dispatchEvent(new Event("mouseout"));
        }
    }
}
```

Reply:
```js
        } else if (previewMode === "reset") {
            clearTimeout(this.hoverTimeoutId);
            this.$target[0].dispatchEvent(new Event("mouseout"));
        }
 ```
It works well without that, so I will leave it like that. In fact, I realized that the `hoverEffectName === this.$target[0].dataset.hoverEffect` in the `if` has fixed the main issue we had with the effect quickly showing.